### PR TITLE
ref(TS) Improve TS

### PR DESCRIPTION
**Pull request by robertpin**
Remove unnecessary @ts-ignores
Use @ts-expect-error instead os @ts-ignore for external dependecies

### CODE REVIEW

**Review by horymury — CHANGES_REQUESTED** [review]

File: react/features/base/net-info/NetworkInfoService.native.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ export default class NetworkInfoService extends EventEmitter {
     /**
      * Stores the native subscription for future cleanup.
      */
-    _subscription: NetInfoSubscription;
+    _subscription: NetInfoSubscription | undefined;
```
[the remark this answers is not in the document]
Comment (robertpin) [reply react/features/base/net-info/NetworkInfoService.native.ts:18]: Hmm, I was certain that `?:` didn't work in classes but it looks like it does. I'll update it.

File: react/features/dropbox/functions.native.ts [hunk 2]
```diff
@@ -70,8 +68,7 @@ export function getSpaceUsage(token: string, _appKey?: any) {
  * @returns {boolean}
  */
 export function isEnabled(state: IReduxState) {
-    const { dropbox = {} } = state['features/base/config'];
+    const { dropbox = { appKey: undefined } } = state['features/base/config'];
```
[the remark this answers is not in the document]
Comment (robertpin) [reply react/features/dropbox/functions.native.ts:71]: It is implicit in JS. But in TS when we try to use `dropbox.appKey` (below this) we get an error saying `appKey does not exist in type {}`. So I added this to make TS happy

File: react/features/filmstrip/actions.native.ts [hunk 3]
```diff
@@ -24,7 +21,7 @@ export function setTileViewDimensions() {
         const { clientHeight: height, clientWidth: width, safeAreaInsets = {} } = state['features/base/responsive-ui'];
         const { left = 0, right = 0, top = 0, bottom = 0 } = safeAreaInsets;
         const columns = getColumnCount(state);
-        const rows = Math.ceil(participantCount / columns);
+        const rows = Math.ceil(participantCount / columns); // @ts-ignore
```
[the remark this answers is not in the document]
Comment (robertpin) [reply react/features/filmstrip/actions.native.ts:24]: The problem is that `StyleSheet` is used in a lot of places. If I change `StyleSheet` we're gonna have a cascade of errors, so I want to treat it separately (have a PR where I change this type and fix all the problems)

File: react/features/filmstrip/components/web/Thumbnail.tsx [hunk 4]
```diff
@@ -1306,9 +1306,17 @@ function _mapStateToProps(state: IReduxState, ownProps: any): Object {
...
+                thumbnailSize: {
+                    height: undefined,
+                    width: undefined
+                }
             },
-            screenshareFilmstripDimensions
+            screenshareFilmstripDimensions = {
+                thumbnailSize: {
+                    height: undefined,
+                    width: undefined
```
[the remark this answers is not in the document]
Comment (robertpin) [reply react/features/filmstrip/components/web/Thumbnail.tsx:1317]: Same as https://github.com/jitsi/jitsi-meet/pull/13282#discussion_r1188200379

Only needed to make TS happy

File: react/features/filmstrip/functions.native.ts [hunk 5]
```diff
@@ -230,7 +227,7 @@ export function getFilmstripDimensions({
     };
     localParticipantVisible?: boolean;
 }) {
-    const { height, width, margin } = styles.thumbnail;
+    const { height, width, margin } = styles.thumbnail; // @ts-ignore
     const conferenceBorder = conferenceStyles.conference.borderWidth || 0;
```
[the remark this answers is not in the document]
Comment (robertpin) [reply react/features/filmstrip/functions.native.ts:231]: Will do in a separate PR

**Review by horymury — APPROVED** [review]

**Pull request merged**